base/loading.jl

```diff
@@ -2600,6 +2600,8 @@ end
 
 function replace_depot_path(path::AbstractString)
     for depot in DEPOT_PATH
+        # Remove trailing slashes in `DEPOT_PATH` if they exist.
+        depot = rstrip(depot, only(Filesystem.pathsep()))
```

Comment:
Maybe something like https://github.com/JuliaLang/julia/blob/85f4db2eaa15fedf738f8a8c16c7168024780f7c/base/file.jl#L235 is somewhat more idiomatic? See also #43137